pinot-spi/src/main/java/org/apache/pinot/spi/filesystem/PinotFS.java

```diff
@@ -108,7 +108,16 @@ public boolean move(URI srcUri, URI dstUri, boolean overwrite)
       // ensures the parent path of dst exists.
       try {
         Path parentPath = Paths.get(dstUri.getPath()).getParent();
-        URI parentUri = new URI(dstUri.getScheme(), dstUri.getHost(), parentPath.toString(), null);
+        /**
+         * Use authority instead of host if the value contains "_": uri.getHost() will be null
+         * This is related to https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6587184
+         * @see java.net.URI.Parser#parseHostname
+         */
+        String host = dstUri.getHost();
+        if (host == null) {
+          host = dstUri.getAuthority();
+        }
+        URI parentUri = new URI(dstUri.getScheme(), host, parentPath.toString(), null, null);
```

Comment:
It seems you're using the constructor with 5 parameters while the current one only has 4 params. It'd be good to clarify that.